source/PluginHelper.cpp

```diff
@@ -0,0 +1,211 @@
+/* PluginHelper.cpp
+Copyright (c) 2023 by RisingLeaf(https://github.com/RisingLeaf)
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "PluginHelper.h"
+
+#if defined _WIN32
+#define WIN32_LEAN_AND_MEAN
+#endif
+
+#include <archive.h>
+#include <archive_entry.h>
+#include <cstring>
+#include <curl/curl.h>
+#include <sys/stat.h>
+#include <stdio.h>
+
+#if defined _WIN32
+#include "text/Utf8.h"
+#include <windows.h>
+#else
+#include <unistd.h>
+#endif
+
+using namespace std;
+
+
+
+namespace PluginHelper {
+	const int MAX_SIZE = 1000000000;
+	bool curlInitialized = false;
+
+	size_t WriteData(void *ptr, size_t size, size_t nmemb, FILE *stream) {
+		size_t written = fwrite(ptr, size, nmemb, stream);
```

Comment:
Currently you're writing the zip to disk and then deleting it later. Instead, you could store the zip in memory and extract it from memory too. Thoughts?

Reply:
For bigger plugins that would be quite the memory load, if we say the max size is a gigabyte, that would mean that we use another gigabyte of memory, if we want to support weaker machines like, just as an example a raspi with only 2GB memory that would be too much imo